deps.nix

```diff
@@ -0,0 +1,192 @@
+# file generated from go.mod using vgo2nix (https://github.com/adisbladis/vgo2nix)
```

Comment:
By using nix to maintain dependencies, I understand that it feels more nix-friendly and pure but do you think that it is more robust than just keeping dependencies with `go.mod`?

Reply:
In order to ensure purity of the build, we have to use [fetchers](https://nixos.org/nixpkgs/manual/#chap-pkgs-fetchers) from `<nixpkgs>`. To my knowledge there is no fetcher which does understand `go.mod` directly. I am unsure about the exact reasons for this but I do think `go.mod` was not considered sufficient to ensure a purity by nixpkgs. Maybe there is some loophole somewhere which could break purity.

This is why we have to convert `go.mod` into something `nixpkgs` understands.